From a7b1c9e6303e4d7185b97b4e924969047e8aadb0 Mon Sep 17 00:00:00 2001
From: TheAssassin <theassassin@assassinate-you.net>
Date: Thu, 28 Feb 2019 20:02:13 +0100
Subject: [PATCH] Install and ship man page

Thanks to Scarlett Moore (@ScarlettGatelyMoore), who kindly provided
the initial version.
---
 resources/AppImageLauncher.1.in | 35 +++++++++++++++++++++++++++++++++
 resources/CMakeLists.txt        | 20 ++++++++++++++++++-
 2 files changed, 54 insertions(+), 1 deletion(-)
 create mode 100644 resources/AppImageLauncher.1.in

diff --git a/resources/AppImageLauncher.1.in b/resources/AppImageLauncher.1.in
new file mode 100644
index 0000000..f179ea9
--- /dev/null
+++ b/resources/AppImageLauncher.1.in
@@ -0,0 +1,35 @@
+.\" Manpage for AppImageLauncher.
+.\" Contact theassassin@assassinate-you.net to correct errors or typos.
+.TH man 1 "28 February 2019" "@APPIMAGELAUNCHER_VERSION@" "AppImageLauncher man page"
+.SH NAME
+AppImageLauncher \- Desktop integration helper for AppImages, for use by Linux distributions.
+.SH SYNOPSIS
+AppImageLauncher [PATH]
+.SH DESCRIPTION
+Helper application for Linux distributions serving as a
+kind of "entry point" for running and integrating AppImages.
+
+AppImage provides a way for upstream developers to provide
+“native” binaries for Linux users just the same way they could
+do for other operating systems. It allows for packaging applications
+for any common Linux based operating system, e.g., Debian.
+AppImages come with all dependencies that cannot be assumed
+to be part of each target system in a recent enough version and
+will run on most Linux distributions without further modifications.
+
+AppImageLauncher makes your Linux desktop AppImage ready™.
+You can integrate AppImages with a single mouse click, and manage
+them from your application launcher. Install AppImageLauncher today
+for your distribution and enjoy using AppImages as easy as never before!
+.SH OPTIONS
+.IP --appimagelauncher-help
+Display this help and exit
+.IP --appimagelauncher-version
+Display version and exit
+.SH ARGUMENTS
+Path to AppImage (mandatory)
+.SH BUGS
+No known bugs.
+.SH AUTHOR
+TheAssassin <theassassin@assassinate-you.net>
+
diff --git a/resources/CMakeLists.txt b/resources/CMakeLists.txt
index 7a4000a..a386b64 100644
--- a/resources/CMakeLists.txt
+++ b/resources/CMakeLists.txt
@@ -25,7 +25,7 @@ install(

 set(INSTALL_MAINTAINER_SCRIPTS OFF CACHE BOOL "")

-# install maintainer scripts for
+# install maintainer scripts that distro packagers might find useful
 if (INSTALL_MAINTAINER_SCRIPTS)
     message(STATUS "Installing maintainer scripts")

@@ -47,3 +47,21 @@ if (INSTALL_MAINTAINER_SCRIPTS)
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/appimagelauncher/maintainer-scripts
     )
 endif()
+
+
+# install man page
+set(INSTALL_MAN_PAGE ON CACHE BOOL "")
+
+if(INSTALL_MAN_PAGE)
+    set(configured_man_page_path ${PROJECT_BINARY_DIR}/cmake/man/AppImageLauncher.1)
+
+    configure_file(
+        ${PROJECT_SOURCE_DIR}/resources/AppImageLauncher.1.in
+        ${configured_man_page_path}
+    )
+
+    install(
+        FILES ${configured_man_page_path}
+        DESTINATION share/man/man1 COMPONENT APPIMAGELAUNCHER
+    )
+endif()
